### PR TITLE
fix(admin) return empty array for enabled_in_cluster property

### LINKED ISSUE #2982: enabled_in_cluster should be an empty JSON array
### Summary

`enabled_in_cluster` field in `GET /` should return an empty JSON array if there are no plugins configured. Currently, returns an empty JSON Object instead.

### Steps To Reproduce

1. Start kong
2. GET / on admin API port

The `plugins.enabled_in_cluster` field is a JSON object if there are no plugins configured in Kong.
```json
.
.
"plugins": {
        "enabled_in_cluster": {},
        "available_on_server": 
.
.

```
But it returns an array of strings if there is a single (or multiple) plugins present:
```json
.
.
"plugins": {
        "enabled_in_cluster": [
            "rate-limiting"
        ],
        "available_on_server": 
.
.

```

This should be consistent and return an empty array in the former case as (de)serialization gets more complicated for an SDK library.

### Additional Details & Logs

- Kong version 0.11.0
- Operating System macOs

### PR DESCRIPTION
**Pull request by hbagdi**
### Summary

Fix admin API `GET /` enabled_in_cluster property to return an empty array when applicable

### Issues resolved

Fix #2982

### CODE REVIEW

File: kong/api/routes/kong.lua [hunk 1]
```diff
@@ -15,7 +16,7 @@ local lua_version = jit and jit.version or _VERSION
 return {
   ["/"] = {
     GET = function(self, dao, helpers)
-      local distinct_plugins = {}
+      local distinct_plugins = cjson.empty_array
```
Comment (thibaultcha) [inline kong/api/routes/kong.lua]: Did you run a test for this (that is, on a non-empty database)? `cjson.empty_array` is a lightuserdata, which should throw an error when we execute `distinct_plugins[#distinct_plugins+1] = row.name` down below. We should probably use `cjson.empty_array_mt` here instead.

We should also introduce a regression test for this.

Comment (hbagdi) [reply kong/api/routes/kong.lua]: Good catch.

`cjson.empty_array_mt` fixes it.

I'll try to add tests as well.

But, I've another problem now:

What is `enabled_in_cluster` suppose to mean?
I expected that it represents plugins which are enabled on some api/consumer currently.
But, a plugin shows up in this list even after I delete it.

- GET / -> `enabled_in_cluster : []`
- Create rate-limiting plugin
- GET / -> `enabled_in_cluster : ["rate-limiting"]`
- Delete the plugin
- GET / -> `enabled_in_cluster : ["rate-limiting"]`

Comment (hbagdi) [reply kong/api/routes/kong.lua]: From https://getkong.org/docs/0.11.x/admin-api/#retrieve-node-information
enabled_in_cluster: Names of plugins that are enabled/configured. That is, the plugins configurations currently in the datastore shared by all Kong nodes

Comment (hbagdi) [reply kong/api/routes/kong.lua]: Never mind, fixed it.

**Comment by thibaultcha** [comment]
@hbagdi Nice. I think this is still missing a regression test though? As of today, since lua-cjson does not support decoding JSON by inserting an array metatable yet, that would mean evaluating the plain JSON response against a regex/pattern and making sure that we have `"enabled_in_cluster":[]` and not `"enabled_in_cluster":{}`. Would you mind providing such a test as well?

**Comment by thibaultcha** [comment]
Manually merged to master with minor edits. Thank you!

**Pull request closed without merging**